content/blog/2024/grid-masonry-nuance.md

```diff
@@ -0,0 +1,480 @@
+Jen also shows some (less convincing)
+flexbox-based alternatives.
+
+If we do want a flexbox fallback,
+though, it requires a bit more work.
+Since the `grid` display mode isn't going to fail entirely,
+we need to override it using `@supports`:
+
+```css
+/* grid fallback… just leave it */
+.grid-fallback {
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
+  grid-template-rows: masonry;
+}
+
+/* flex fallback, additional definition */
+.flex-fallback {
+  display: flex;
+  flex-wrap: wrap;
+  /* maybe other things? */
+
+  @supports (grid-template-rows: masonry) {
+    display: grid;
+    grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
+    grid-template-rows: masonry;
+  }
+}
+```
+
+I'm not sure why we would want that fallback, exactly.
+Flexbox rows are all vertically aligned,
+just like our default grid rows.
+But I can imagine situations where we want
+a flexbox variants at some screen size,
+to create a different effect.
+
+Using an alternate `masonry` display mode
+means we'll fall back to `block` display
+if we don't do anything else.
+To get the grid fallback
+with columns that match our masonry layout,
+we have to specify both --
+duplicating the track definitions:
+
+```css
+/* grid fallback… duplicate properties */
+.grid-fallback {
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
+
+  display: masonry;
+  masonry-template-tracks: repeat(auto-fit, minmax(200px, 1fr));
+}
+```
+
+We could put the track definition inside a variable if we want,
+but it still requires explicit duplication.
+Because we can rely on various properties working
+in only one layout mode,
+and we get a simple mode toggle,
+it's fairly simple to set up any fallback we want.
+But the trade off is that we don't have
+the the most obvious fallback _built in_ from the start.
+
+While I was experimenting with different fallback paths,
+I was trying to maintain the spirit of masonry in some way --
+tightly packed columns --
+with as little effort as possible.
+My favorite solution was the default fallback
+provided by grid-integrated masonry,
+with the addition of rough aspect-ratio attributes when possible --
+e.g. `is-short` and `is-tall`.
+All items span multiple rows,
+but short items span fewer rows,
+and tall items span more rows.
+
+<figure>
+  {{ embed.codepen(
+    id='wvVKvrN',
+    title='Rough Puff Masonry',
+    user='miriamsuzanne'
+  ) }}
+  <figcaption>
+    I borrowed the markup and images
+    from Jen Simmons,
+    and I'm pretty happy with the result.
+  </figcaption>
+</figure>
+
+It's not a perfect solution,
+but I was surprised how well it works.
+This is a damn good fallback story in my mind.
+Of course, _effort_ here is hard to measure.
+If we know the size of each item,
+and have access to the markup,
+then this is pretty straight-forward.
+But those are big _ifs_.
+
+On my personal site,
+image galleries fit this description perfectly.
+Maybe I'll start using it.
```

Comment:
Should we link to an example?

Reply:
I haven't started using it yet. Or do you just mean, and example gallery that _could be_ masonry-like?

Reply:
I just meant to see what kind of gallery you currently have that could use it. No pressure though, I was just curious